### ISSUE TITLE
void_checks false positive with FutureOr<void> and NNBD

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#2185]
```dart
// @dart = 2.9
import 'dart:async';

void m(FutureOr<void> Function() f) {}

void main() {
  m(() {}); // void_checks false positive
}
```

**Comment by natebosch** [comment]
Here is another example that is easy to hit with `unawaited` from `package:pedantic` (at least until https://github.com/dart-lang/pedantic/pull/69 goes in)

```dart
void main() {
  unawaited(Future.value(1).then((_) {})); // Error on argument to then
}

void unawaited(Future<void> f) {}
```
